### PR TITLE
Fetch invalid profiles and regenerate them if needed

### PR DESCRIPTION
**Pull request by icecrystal23**
Fixes https://github.com/fastlane/fastlane/issues/4220 by finding invalid profiles and regenerating them if needed

### CODE REVIEW

File: sigh/lib/sigh/runner.rb [hunk 1]
```diff
@@ -65,7 +65,7 @@ def profile_type
     # Fetches a profile matching the user's search requirements
     def fetch_profiles
       UI.message "Fetching profiles..."
-      results = profile_type.find_by_bundle_id(Sigh.config[:app_identifier]).find_all(&:valid?)
+      results = profile_type.find_by_bundle_id(Sigh.config[:app_identifier])
```
Comment (KrauseFx) [inline sigh/lib/sigh/runner.rb]: But it still has to be valid for us to download. Even if we repair it, something might go wrong on either fastlane's or Apple's side, so I'd say we should leave this here. It's better to fail early, than later at the build process

Comment (icecrystal23) [inline sigh/lib/sigh/runner.rb]: `sigh` can't repair the profile if it never finds it in the first place
we could double check that its valid before we download it at the end?

Comment (KrauseFx) [inline sigh/lib/sigh/runner.rb]: Oh, right, I didn't think about this.

I'm not yet sure about this change, as it changes the way sigh works for the different use-cases. I'll give it some more thought.

**Comment by KrauseFx** [comment]
I'm actually working on some refactors on how _sigh_ works, as the API has changed. I'll close this PR for now, and let's discuss this again once the new releases for _sigh_ and _spaceship_ are live 👍

**Pull request closed without merging**